litert/vendors/qualcomm/core/transformation/mha_to_sha.cc

```diff
@@ -407,4 +464,239 @@ size_t OptimizeMHADecode(std::function<bool(OpWrapper&)> validate_op_config,
       "[G2G] Validation failed. Rolling back to the original graph.");
   return 1;
 }
+
+size_t OptimizeMHAAttn(std::function<bool(OpWrapper&)> validate_op_config,
+                       std::vector<OpWrapper>& ops, size_t attn_start_index,
+                       TensorPool& tensor_pool, size_t pattern_size) {
+  // Connection check: Reshape -> NotEqual -> Select
+  size_t start_index = attn_start_index;
+  if (!(IS_CONNECTED(kAttnReshape, 0, kAttnNotEqual, 0)) &&
+      (IS_CONNECTED(kAttnNotEqual, 0, kAttnSelect, 0))) {
+    return 1;
+  }
+  auto attn_reshape_op = ops[attn_start_index + kAttnReshape];
+  auto attn_not_equal_op = ops[attn_start_index + kAttnNotEqual];
+  // Count the operations that have NotEqual as their source op.
+  const auto& reshape_in = attn_reshape_op.GetInputTensor(0);
+  size_t num_out = GetOpCountWithInput(ops, QnnOpCode::kElementWiseSelect,
+                                       attn_not_equal_op.GetOutputTensor(0));
+  if (num_out <= 0) {
```

Comment:
`num_out` is unsigned.
```suggestion
  if (num_out == 0) {
```